subprojects/snapshots/src/main/java/org/gradle/internal/vfs/impl/DefaultVirtualFileSystem.java

```diff
@@ -43,16 +44,17 @@
 import java.util.function.Supplier;
 
 public class DefaultVirtualFileSystem implements VirtualFileSystem {
-    private final AtomicReference<FileHierarchySet> root = new AtomicReference<>(FileHierarchySet.EMPTY);
+    private final AtomicReference<FileHierarchySet> root;
     private final Stat stat;
     private final DirectorySnapshotter directorySnapshotter;
     private final FileHasher hasher;
     private final StripedProducerGuard<String> producingSnapshots = new StripedProducerGuard<>();
 
-    public DefaultVirtualFileSystem(FileHasher hasher, Interner<String> stringInterner, Stat stat, String... defaultExcludes) {
+    public DefaultVirtualFileSystem(FileHasher hasher, Interner<String> stringInterner, Stat stat, boolean caseSensitive, String... defaultExcludes) {
         this.stat = stat;
         this.directorySnapshotter = new DirectorySnapshotter(hasher, stringInterner, defaultExcludes);
         this.hasher = hasher;
+        this.root = new AtomicReference<>(DefaultFileHierarchySet.empty(caseSensitive ? CaseSensitivity.CASE_SENSITIVE : CaseSensitivity.CASE_INSENSITIVE));
```

Comment:
Why not take `CaseSensitivity` instead of the `boolean` parameter?

Reply:
Passing `CaseSensitivity` now.